frontend/src/components/common/Admin/RemoveUserModal.tsx

```
@@ -1,33 +1,76 @@
 import {
   Button,
+  Divider,
   Modal,
   ModalContent,
   ModalFooter,
   ModalOverlay,
   useDisclosure,
 } from "@chakra-ui/react";
-import React from "react";
+import React, { useContext } from "react";
+import AuthContext from "../../../contexts/AuthContext";
 import { CloseOutlineIcon } from "../icons";
-import ModalText from "../ModalText";
+import ModalText from "./ModalText";
 import RemoveUserConfirmationModal from "./RemoveUserConfirmationModal";
+import RemoveUserErrorModal from "./RemoveUserErrorModal";
+import { Role } from "../../../types/AuthTypes";
 
 interface RemoveUserModalProps {
   name: string;
   email: string;
   onCloseParent: () => void;
 }
 
+interface UserGQLObject {
+  userByEmail: { id: string };
+}
+
+const GET_USER = gql`
+  query RemoveUserModal_UserByEmail($email: String!) {
+    userByEmail(email: $email) {
+      id
+    }
+  }
+`;
+
+const REMOVE_USER = gql`
+  mutation RemoveUserModal_DeleteUserByEmail($email: String!) {
+    deleteUserByEmail(email: $email)
+  }
+`;
+
 const RemoveUserModal = ({
   name,
   email,
   onCloseParent,
 }: RemoveUserModalProps): React.ReactElement => {
   const { onOpen, onClose, isOpen } = useDisclosure();
   const [showConfirmation, setShowConfirmation] = React.useState(false);
+  const [showError, setShowError] = React.useState(false);
+  const [removeUser, { error }] = useMutation<{ removeUser: null }>(
+    REMOVE_USER,
+  );
+  const [user, setUser] = React.useState<string | null>("");
+  const { authenticatedUser } = useContext(AuthContext);
+
+  useQuery(GET_USER, {
+    variables: { email },
+    fetchPolicy: "cache-and-network",
+    onCompleted: (data: UserGQLObject) => setUser(data.userByEmail.id),
+  });
 
-  const onRemoveUserClick = () => {
-    console.log("Remove user with email: ", email);
-    setShowConfirmation(true);
+  const onRemoveUserClick = async () => {
+    if (user && authenticatedUser?.role === Role.ADMIN) {
```

do we need the user role check? i don't think non-admins have access to this page

---

similar question for `user`, do we need to refetch and check existence?

---

hmm i guess validating if the user exists is not needed given that we are working with a table of emails we populated

---

yeah, plus i think if the user doesn't exist, the BE should throw an error anyway